doc/source/f2py/f2py.getting-started.rst

```diff
@@ -96,98 +99,95 @@ Python the Fortran subroutine ``FIB`` is accessible via ``fib1.fib``::
     subroutine. As a result, any possible changes to the (copy of)
     input array have no effect to the original argument, as
     demonstrated below::
 
-      >>> a = numpy.ones(8, 'i')
+      >>> a = np.ones(8, 'i')
       >>> fib1.fib(a)
       >>> print(a)
       [1 1 1 1 1 1 1 1]
 
-    Clearly, this is not an expected behaviour. The fact that the
-    above example worked with ``dtype=float`` is considered
-    accidental.
+    Clearly, this is not an expected behaviour; especially since Fortran
+    typically passes by reference. The fact that the above example worked with
+    ``dtype=float`` is considered accidental.
 
-    F2PY provides ``intent(inplace)`` attribute that would modify
+    F2PY provides an ``intent(inplace)`` attribute that modifies
     the attributes of an input array so that any changes made by
-    Fortran routine will be effective also in input argument. For example,
-    if one specifies ``intent(inplace) a`` (see below, how), then
-    the example above would read::
+    Fortran routine will be reflected in the input argument. For example,
+    if one specifies the ``intent(inplace) a`` directive (see subsequent
+    sections on how), then the example above would read::
 
-      >>> a = numpy.ones(8, 'i')
+      >>> a = np.ones(8, 'i')
       >>> fib1.fib(a)
       >>> print(a)
       [  0.   1.   1.   2.   3.   5.   8.  13.]
 
-    However, the recommended way to get changes made by Fortran
-    subroutine back to Python is to use ``intent(out)`` attribute. It
-    is more efficient and a cleaner solution.
-
-  * The usage of ``fib1.fib`` in Python is very similar to using
-    ``FIB`` in Fortran. However, using *in situ* output arguments in
-    Python indicates a poor style as there is no safety mechanism
-    in Python with respect to wrong argument types. When using Fortran
-    or C, compilers naturally discover any type mismatches during
-    compile time but in Python the types must be checked in
-    runtime. So, using *in situ* output arguments in Python may cause
-    difficult to find bugs, not to mention that the codes will be less
-    readable when all required type checks are implemented.
-
-  Though the demonstrated way of wrapping Fortran routines to Python
-  is very straightforward, it has several drawbacks (see the comments
-  above).  These drawbacks are due to the fact that there is no way
-  that F2PY can determine what is the actual intention of one or the
-  other argument, is it input or output argument, or both, or
-  something else. So, F2PY conservatively assumes that all arguments
-  are input arguments by default.
-
-  However, there are ways (see below) how to "teach" F2PY about the
-  true intentions (among other things) of function arguments; and then
-  F2PY is able to generate more Pythonic (more explicit, easier to
-  use, and less error prone) wrappers to Fortran functions.
+    However, the recommended way to get changes made by Fortran subroutine
```

Comment:
"made by a Fortran"

Reply:
Perhaps 

>However, the recommended way to have changes made by Fortran subroutine
    propagate to Python is to use the ``intent(out)`` attribute

Reply:
Looks good.